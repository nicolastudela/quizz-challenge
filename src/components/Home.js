import React from "react";
import { Typography } from "@material-ui/core";

export const Home = ({user, ...rest}) => (
  <>
    <Typography variant="h3" align="center">{`Welcome ${user.name}`}</Typography>
  </>
);

export default Home
