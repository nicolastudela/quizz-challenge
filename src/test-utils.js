/* eslint-disable react/prop-types */
import React from "react";
import { ThemeProvider } from "@material-ui/core/styles";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import theme from "./theme";

export const INITIAL_EMPTY_STORE = {
  user: {},
  questionnaire: {
    items: [],
    isFetching: false,
    error: null
  }
};

const mockStore = configureStore([thunk]);

export const BasicTestWrapper = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </ThemeProvider>
  );
};

export const WithReduxStoreTestWrapper = ({
  children,
  store = INITIAL_EMPTY_STORE,
  initialEntries,
  initialIndex
}) => {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter initialEntries={initialEntries} initialIndex={initialIndex}>
        <Provider store={mockStore(store)}>{children}</Provider>
      </MemoryRouter>
    </ThemeProvider>
  );
};
