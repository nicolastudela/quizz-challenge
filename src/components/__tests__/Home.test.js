import React from "react";
import renderer from "react-test-renderer";
import { BasicTestWrapper } from "test-utils";
import { Home } from "../Home";

describe("Home page", () => {
  it("renders propery", () => {
    expect.assertions(1);
    const component = renderer.create(
      <BasicTestWrapper>
        <Home />
      </BasicTestWrapper>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
