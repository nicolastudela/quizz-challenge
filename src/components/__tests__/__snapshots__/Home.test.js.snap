// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders propery 1`] = `
Array [
  <h3
    className="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
  >
    Welcome undefined
  </h3>,
  <div
    className="MuiBox-root MuiBox-root-32"
  />,
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
  >
    Your last results:
  </h4>,
  <nav
    aria-label="trivia-results"
    className="MuiList-root MuiList-padding"
  />,
  <div
    className="MuiBox-root MuiBox-root-37"
  />,
  <div
    className="MuiBox-root MuiBox-root-38"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "width": "30%",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Reset User
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "width": "50%",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Play Again?
      </span>
    </button>
  </div>,
]
`;
